### PR TITLE
Heedl 606 hide inactive admin details

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-606

### Description
Change various places where we access admins, and either filter out inactive admins or add an extra "(inactive)" into the display

### Screenshots
View delegate course example:
image

Group Courses example:
image

Role Profiles example:
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml [hunk 1]
```diff
@@ -1,207 +1,216 @@
...
+
+  </div>
+  @if (Model.CandidateAssessmentSupervisorVerificationSummaries.Any()) {
+    <div class="nhsuk-summary-list__row">
+      <dt class="nhsuk-summary-list__key">
+        Response verifiers
+      </dt>
+      <dd class="nhsuk-summary-list__value">
+        @foreach (var candidateAssessmentSupervisorVerificationSummary in Model.CandidateAssessmentSupervisorVerificationSummaries) {
+          <span>@DisplayStringHelper.GetPotentiallyInactiveAdminName(candidateAssessmentSupervisorVerificationSummary.Forename, candidateAssessmentSupervisorVerificationSummary.Surname, candidateAssessmentSupervisorVerificationSummary.AdminActive) (@candidateAssessmentSupervisorVerificationSummary.Email) - @candidateAssessmentSupervisorVerificationSummary.VerifiedCount</span>
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml:107]: A pretty nasty line. Worth adding the text content to the view model?

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Seems mostly good from what I've looked at. Few questions lying around.

I left a few comments on the SelfAssessmentDataService before it was agreed to leave those areas for Kevin to look into. I'll leave them there for now, but I wouldn't be surprised if they become irrelevant.

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/CourseDelegates/DelegateProgressViewModel.cs [hunk 2]
```diff
@@ -46,34 +55,12 @@ public class DelegateProgressViewModel : DelegateCourseInfoViewModel
         }
 
         public DelegateProgressAccessRoute AccessedVia { get; set; }
-        public int CustomisationCentreId { get; set; }
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/CourseDelegates/DelegateProgressViewModel.cs:49]: I've cleaned up and simplified the name properties using the new name format helpers, then deleted some unused properties.

File: DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs [hunk 3]
```diff
@@ -146,9 +147,10 @@ public IEnumerable<Group> GetGroupsForCentre(int centreId)
                         GroupDescription,
                         (SELECT COUNT(*) FROM GroupDelegates AS gd WHERE gd.GroupID = g.GroupID) AS DelegateCount,
                         ({CourseCountSql}) AS CoursesCount,
-                        g.CreatedByAdminUserID As AddedByAdminId,
-                        au.Forename AS AddedByFirstName,
-                        au.Surname AS AddedByLastName,
+                        g.CreatedByAdminUserID AS AddedByAdminId,
+                        CASE WHEN au.Active = 1 THEN au.Forename ELSE NULL END AS AddedByFirstName,
+                        CASE WHEN au.Active = 1 THEN au.Surname ELSE NULL END AS AddedByLastName,
+                        au.Active AS AddedByAdminActive,
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/DataServices/GroupsDataService.cs:153]: Ah whoops I missed this one. This is incorrect behaviour from Martin

File: DigitalLearningSolutions.Data/Models/DelegateGroups/Group.cs [hunk 4]
```diff
@@ -26,8 +28,6 @@ public class Group : BaseSearchableItem
 
         public bool ChangesToRegistrationDetailsShouldChangeGroupMembership { get; set; }
 
-        public string AddedByName => $"{AddedByFirstName} {AddedByLastName}";
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/Models/DelegateGroups/Group.cs:29]: No, that's kind of the point of getting rid of these. Changing them to display strings is the domain of the Web project (or whatever frontend is applied above the services). e.g. what if a different page wanted "Surname, Forename" with the same Group class?

File: DigitalLearningSolutions.Data/DataServices/UserDataService/AdminUserDataService.cs [hunk 5]
```diff
@@ -7,45 +7,49 @@
...
-                        au.IsFrameworkContributor,
-                        au.IsWorkforceManager,
-                        au.IsWorkforceContributor,
-                        au.IsLocalWorkforceManager,
-                        au.ImportOnly,
-                        au.FailedLoginCount
-                    FROM AdminUsers AS au
-                    INNER JOIN Centres AS ct ON ct.CentreID = au.CentreID
-                    LEFT JOIN CourseCategories AS cc ON cc.CourseCategoryID = au.CategoryID
+                @$"{BaseSelectAdminQuery}
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/AdminUserDataService.cs:52]: I've had a scan up the calling code for them and I believe they are, but I may have missed some

File: DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs [hunk 6]
```diff
@@ -36,6 +36,7 @@ public partial class UserDataService
                                                             AND au.Password = cd.Password
                                                                 AND au.CentreID = cd.CentreID
                                                                 AND au.Email != ''
+                                                                AND au.Active = 1
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs:39]: At least when we currently promote a delegate to admin, or register a new centre manager account, we're saving the same password hash. 
@SteveJacksonSoft Is this an issue we need to look into?

File: DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml (hunk 1 above)
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Web/Views/Supervisor/SignOffProfileAssessment.cshtml:107]: There isn't a mapping to a ViewModel here, and I've prefered the minimal approach for these pages made by HEE.

**Review by JBsoftwire — CHANGES_REQUESTED** [review]
No major new comments but a couple of questions.

File: DigitalLearningSolutions.Data/Services/SupervisorService.cs [hunk 7]
```diff
@@ -1,741 +1,958 @@
-﻿namespace DigitalLearningSolutions.Data.Services
```
[the remark this answers is not in the document]
Comment (AlexJacksonDS) [reply DigitalLearningSolutions.Data/Services/SupervisorService.cs]: Yeah there are a few changes, but the plan is to get Kevin to look over these files

File: DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs:39]: Good spot. The only scenario I can think of in which corresponding admin and delegate accounts would have different passwords is if they were made in the old system. I'm not sure if that ensured they had the same password or not but I'll check.

[the remark this answers is not in the document]
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs:39]: Looks like they won't necessarily have the same password hash...

[the remark this answers is not in the document]
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs:39]: I'll tell Kevin, probably not a big enough issue to worry, and easy to fix after the user account refactor.

[the remark this answers is not in the document]
Comment (SteveJacksonSoft) [reply DigitalLearningSolutions.Data/DataServices/UserDataService/DelegateUserCardDataService.cs:39]: He's not bothered.

File: DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/AddSupervisorSummary.cshtml [hunk 8]
```diff
@@ -27,7 +28,7 @@
       Supervisor
     </dt>
     <dd class="nhsuk-summary-list__value">
-      @Model.Supervisor.Forename @Model.Supervisor.Surname (@Model.Supervisor.Email)
+      @DisplayStringHelper.GetPotentiallyInactiveAdminName(Model.Supervisor.Forename, Model.Supervisor.Surname, Model.Supervisor.Active) (@Model.Supervisor.Email)
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/Views/LearningPortal/SelfAssessments/AddSupervisorSummary.cshtml:31]: The data service that retrieved possible supervisors to add includes `WHERE Active = 1` so this helper method is unlikely to be needed, though I suppose it's not doing any harm.

File: DigitalLearningSolutions.Web.Tests/Helpers/DisplayStringHelperTests.cs [hunk 9]
```diff
@@ -182,6 +182,46 @@ public void GetSortableFullName_returns_correctly_formatted_name_with_email_when
             result.Should().Be("FirstName LastName (email@email.com)");
         }
 
+        [Test]
+        public void GetNameWithEmailForDisplay_returns_correctly_formatted_name_for_active_admin()
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web.Tests/Helpers/DisplayStringHelperTests.cs:186]: This name for this test, and the two new ones after it, don't match the method named used.

File: DigitalLearningSolutions.Data/Models/Courses/DelegateCourseInfo.cs [hunk 10]
```diff
@@ -28,6 +29,7 @@ public class DelegateCourseInfo : CourseNameInfo
             int? enrolledByAdminId,
             string? enrolledByForename,
             string? enrolledBySurname,
+            bool enrolledByAdminActive,
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Data/Models/Courses/DelegateCourseInfo.cs:32]: Should also be nullable.

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Couple of minor points that should be quick to fix, and some merge conflicts to sort, but otherwise looks good I think.

**Review by DanBloxham-sw — APPROVED** [review]
I think that covers everything.

**Pull request merged**